pkg/gofr/datasource/file/s3/fs.go

```diff
@@ -20,9 +18,23 @@ import (
 	file "gofr.dev/pkg/gofr/datasource/file"
 )
 
+const (
+	TypeFile      = "file"
+	TypeDirectory = "directory"
+)
+
+var (
+	ErrIncorrectFileType = errors.New("incorrect file type")
+)
+
+// Conn struct embeds the *ftp.ServerConn returned by ftp server on successful connection.
+type conn struct {
+	*s3.Client
+}
+
 type fileSystem struct {
 	s3file
-	conn    *s3.Client
+	conn    s3conn
```

Comment:
You define conn, then use s3conn

Maybe you could rename conn to s3Client or s3ClientConn

Reply:
Yes, anything that not increase the confusion like having s3conn and conn being different things